### DISCUSSION TITLE
RuntimeError: Indicator "OBV(C,V)" errored with exception: input array type is not double

### DISCUSSION THREAD

**Discussion by ghost** [kernc/backtesting.py#481] [Q&A]
Hello, 

I am having the following problem with my code
```py
from backtesting import Backtest, Strategy
from talib import ADX, PLUS_DI, MINUS_DI,OBV,SMA
from backtesting.test import GOOG,EURUSD

class ADX_Strategy(Strategy):

    def init(self):
        self.High = self.data.High
        self.Close = self.data.Close
        self.Low = self.data.Low
        self.Volume = self.data.Volume

        self.ADX = self.I(ADX, self.High,self.Low,self.Close)
        self.PLUS_DI = self.I(PLUS_DI,self.High,self.Low,self.Close)
        self.MINUS_DI = self.I(MINUS_DI,self.High,self.Low,self.Close)
        self.OBV = self.I(OBV,self.Close,self.Volume)

    def next(self):

        print("Test")
        pass`
```
And it is throwing the following exception in the run() call:
> ```
> ipython-input-40-d7cee23cbdc2> in init(self)
>      10         self.PLUS_DI = self.I(PLUS_DI,self.High,self.Low,self.Close)
>      11         self.MINUS_DI = self.I(MINUS_DI,self.High,self.Low,self.Close)
> ---> 12         self.OBV = self.I(OBV,self.Close,self.Volume)
>      13 
>      14     def next(self):
> 
> ~/anaconda3/envs/Mamani_1_0_0/lib/python3.8/site-packages/backtesting/backtesting.py in I(self, func, name, plot, overlay, color, scatter, *args, **kwargs)
>     126             value = func(*args, **kwargs)
>     127         except Exception as e:
> --> 128             raise RuntimeError(f'Indicator "{name}" errored with exception: {e}')
>     129 
>     130         if isinstance(value, pd.DataFrame):
> 
> RuntimeError: Indicator "OBV(C,V)" errored with exception: input array type is not double
> ```

Any idea what could be wrong? 

Thanks,
Carlos Mamani

**Comment by ghost** [comment]
Edit:
I had to convert the Volume Series type to Double, for some reason it fails when I call the OBV indicator through the backtesting.Indicator class, but If I call the OBV class directly from the talib module it doesn't fail.

**Comment by kernc** [comment]
FWIK, ta-lib functions all expect `array.astype(float)` as their inputs when of `np.ndarray` type. The reason you see it works with plain `OBV(df.Close, df.Volume)` is that [ta-lib does this on-the-fly `pd.Series` unwrapping](https://github.com/mrjbq7/ta-lib/blob/ffc63aba9c899682fcc74bd79f940ddeb397c0d7/talib/__init__.py#L29-L35).

You can, therefore, call:
```py
        self.OBV = self.I(OBV, self.Close, self.Volume.astype(float))
```
or, using the [`.s` accessor](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data):
```py
        self.OBV = self.I(OBV, self.Close, self.Volume.s)
```
